airbyte-cdk/bulk/core/base/src/testFixtures/kotlin/io/airbyte/cdk/extensions/JunitTimeoutInvocationInterceptor.kt

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2025 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.extensions
+
+import java.lang.reflect.Method
+import java.time.Duration
+import java.time.format.DateTimeParseException
+import java.util.Collections
+import java.util.HashMap
+import java.util.concurrent.TimeUnit
+import java.util.regex.Pattern
+import org.junit.jupiter.api.Assertions.assertTimeoutPreemptively
+import org.junit.jupiter.api.extension.ExtensionContext
+import org.junit.jupiter.api.extension.InvocationInterceptor
+import org.junit.jupiter.api.extension.ReflectiveInvocationContext
+
+class JunitTimeoutInvocationInterceptor : InvocationInterceptor {
+    @Throws(Throwable::class)
+    override fun interceptTestMethod(
+        invocation: InvocationInterceptor.Invocation<Void>,
+        context: ReflectiveInvocationContext<Method>,
+        extensionContext: ExtensionContext
+    ) {
+        assertTimeoutPreemptively<Void>(
+            getTimeout(),
+        ) {
+            invocation.proceed()
+        }
+    }
+
+    companion object {
```

Comment:
everything in this companion object is just copied from LoggingInvocationInterceptor. Happy to make edits, but it's basically just string munging to parse a string into a duration.